app/services/reports_tar_reader.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rubygems/package'
+require 'zlib'
+
+# Finds the files in a insights tar and returns them as an array
+class ReportsTarReader
+  def initialize(file)
+    @file = file
+  end
+
+  def reports
+    tar_extract = Gem::Package::TarReader.new(Zlib::GzipReader.open(@file))
+    tar_extract.rewind # The extract has to be rewinded after every iteration
+    files = []
+    tar_extract.each do |entry|
+      files << entry.read
+    end
+    tar_extract.close
```

Comment:
This seems like a great opportunity to use `yield` and wrap all this TarReader stuff in its own method which can contain the `ensure` to close the archive.